src/main/java/picard/illumina/BarcodeMetric.java

```diff
@@ -0,0 +1,128 @@
+package picard.illumina;
+
+import htsjdk.samtools.metrics.MetricBase;
+
+/**
+ * Metrics produced by the ExtractIlluminaBarcodes program that is used to parse data in
+ * the basecalls directory and determine to which barcode each read should be assigned.
+ */
+public class BarcodeMetric extends MetricBase {
```

Comment:
Since this is a breaking change, perhaps leave the inner class in `ExtractIlluminaBarcodes` and deprecate it?  Likely it will look ugly in the online documentation, but I know that myself I both rely on this class `ExtractIlluminaBarcodes` and point folks to its documentation.